torchao/quantization/observer.py

```diff
@@ -16,14 +16,35 @@
 
 @dataclass(frozen=True)
 class GranularityType:
+    """
+    Base class for representing the granularity of quantization.
+
+    This class serves as a parent for specific granularity types used in 
+    quantization operations, such as per-tensor or per-axis quantization.
+    """
     pass
 
 @dataclass(frozen=True)
 class PerTensor(GranularityType):
+    """
+    Represents per-tensor granularity in quantization.
+
+    This granularity type calcualtes the quantization parameters
+    based off the entire tensor.
+    """
     pass
 
 @dataclass(frozen=True)
 class PerAxis(GranularityType):
+    """
+    Represents per-axis granularity in quantization.
+
+    This granularity type calcualtes different quantization parameters
+    along a specified axis of the tensor.
+
+    Attributes:
+        axis (int): The axis along which reduction is performed.
+    """
     axis: int
```

Comment:
@jerryzh168 
This kinda works backwards to what I would think

e.g with tensor:
9x2048

If I was calculating with axis wise and axis = 0, my first thought would be that this would produce min_values
1x2048 e.g. along the columns and vice versa  for axis=1 but it seems that this is flipped

Reply:
axis = 0 means one pair of qparams per element in axis = 0, in this case it would be 9 qparams.

this is consistent with the `ch_axis` arg we have before in: https://github.com/pytorch/pytorch/blob/f2a7228aed15f97cd4389b4240fa24e25b5eeb4f/torch/ao/quantization/observer.py#L690 and https://github.com/pytorch/pytorch/blob/7b280c31bac252c7741067a68b1675aefeb1da35/aten/src/ATen/native/native_functions.yaml#L7539